### PR TITLE
⚙️ DEVELOP

### PR DESCRIPTION
**Pull request by xat**
# refactored startpage, loading of settings and user-profile page

moved loading of initial data into use-settings hook

declare routes directly in App component

started refactoring search component. exploring patterns/abstractions

further refactorings on the search component

added graphql dependency

re-added subjkt search-results

small fixes after rebase

always activate infinite scroll

added tezospride feed

refactored language handling, removed unused code, fixed some of the broken ui sections

started refactoring of user-profile page (WIP)

fixes after rebase (still a lot of broken stuff)

fixed redirects that happen on sync page

fixed logo handling

reformat after rebase

fetch metadata from indexer instead of from ipfs (like it's the case upstream)

re-added subjkt support

prefix the feed-related URIs with '/feed', so that they don't conflict with subjkt URIs

started to re-add tab logic on user profile page (WIP)

fixes after rebase

fixed routing on profile page

re-implemented creations-tab on profile-page (filter logic still missing)

re-added filter-logic in creations tab

re-added collection tab on user-profile page (WIP)

re-implemented filter logic on collections tab of user-profile page

fixed profile-page header

fixed collab-tab on user-profile page

fixed feed routes on startpage

fixed collab view

fixes after rebase

### CODE REVIEW

**Comment by xat** [comment]
this branch contains the refactorings i did over the last months. these were the things done:

- all the initial settings now get loaded in the `use-settings` hook.
- react-router was updated to `v6`
- refactored the start page and the user profile page to follow a more declarative pattern

keep in mind that:

- there are some places in the code where blocked wallets still need to be filtered out (see TODOs)
- in the user profile page the active state of the menu is a bit messed up
- i removed the hdao stuff on the user profile page. it needs to be decided if it should be re-added.
- i also removed the ui that was responsible for canceling v1 swaps from the user profile page

this also needs a bunch of manual testing. imo, there is probably still a lot more that could be refactored.

File: src/components/header/index.js [hunk 1]
```diff
@@ -18,9 +18,25 @@ import styles from './styles.module.scss'
 import { getItem, setItem } from '../../utils/storage'
 import { EventBanner } from '@components/event-banner/index'
 import { useWindowScroll } from 'react-use'
+import useSettings from '@hooks/use-settings'
+/* import { BeaconWallet } from '@taquito/beacon-wallet'
```
Comment (xat) [inline src/components/header/index.js]: remove the commented out code-block?

File: src/components/header/index.js [hunk 2]
```diff
@@ -29,7 +45,7 @@ export const Header = () => {
   useEffect(() => {
     context.setAccount()
     context.setTheme(getItem('theme') || setItem('theme', 'dark'))
-    context.setLogo()
+    //context.setLogo()
```
Comment (xat) [inline src/components/header/index.js]: remove

File: src/data/hicdex.js [hunk 3]
```diff
@@ -142,7 +142,7 @@ shareholder(where: {holder_id: {_eq: $address}, holder_type: {_eq: "core_partici
 }`
 
 export const getCollabTokensForAddressesByShare = `query GetCollabTokens($addresses: [String!]) {
-  split_contract(where: {shareholder: {holder_id: {_in: $addresses}}, contract: {tokens: {supply: {_gt: "0"}, _and: {is_signed: {_eq: true}}}}}) {
+  split_contract(where: {shareholder: {holder_id: {_in: [KT1DWnLiUkNtAQDErXxudFEH63JC6mqg3HEx]}}, contract: {tokens: {supply: {_gt: "0"}, _and: {is_signed: {_eq: true}}}}}) {
```
Comment (xat) [inline src/data/hicdex.js]: wondering if this is correct...

Comment (Zir0h) [reply src/data/hicdex.js]: I don't think so. The $address var has to be used, or hasura will whine about it.
Also KT1DWnLiUkNtAQDErXxudFEH63JC6mqg3HEx should be "KT1DWnLiUkNtAQDErXxudFEH63JC6mqg3HEx".
Maybe this was hard coded for debugging purposes once?

Comment (Zir0h) [reply src/data/hicdex.js]: Nevermind, it "works" but it throws some warnings in the console :smile:

Comment (xat) [reply src/data/hicdex.js]: I think I added it indeed for debugging reasons

File: src/App.js [hunk 4]
```diff
@@ -1,35 +1,56 @@
...
+        <Route path="/sync" element={<Sync />} />
+        <Route path="/mint" element={<Mint />} />
+        <Route path="/collaborate" element={<Collaborate />} />
+        <Route path="/objkt/:id" element={<ObjktDisplay />} />
+        <Route path="/galleries" element={<Galleries />} />
+        <Route path="/gallery/:id" element={<GalleryDetail />} />
+        <Route path="/config" element={<Config />} />
+        <Route path="/tags/:id" element={<Tags />} />
+        <Route path="/:id/:collection?" element={<Display />} />
+        <Route path="/:subjkt" element={<Display />} />
```
Comment (xat) [inline src/App.js]: can maybe be removed

File: src/hooks/use-settings.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+  )
+}
+
+async function fetchSettings() {
+  const [
+    objktBlockListResponse,
+    banBlockListResponse,
+    logosResponse,
+    logosPrideResponse,
+    reiaRestrictedListResponse,
```
Comment (xat) [inline src/hooks/use-settings.js]: `reia` => `teia`

**Comment by xat** [comment]
https://teia-ui-pr-192.onrender.com/collab/horrorgods

links to objkt detail pages contain `undefined` as objkt_id.

**Comment by xat** [comment]
@melMass can you remind me what the problem with the feed cache was?

i remember that i fixed a problem like that a few weeks ago that was reported by @merchantcoppola

**Comment by melMass** [comment]
> @melMass can you remind me what the problem with the feed cache was?
> 
> i remember that i fixed a problem like that a few weeks ago that was reported by @merchantcoppola

Did I ever answer that on Dicsord? Sorry I'm a bit lost recently 😅 !

I'm currently "reviving" my develop PR! The plan is to merge main into it and then back here, does it sound good to you? I saw on discord your proposal of merging develop asap  (within 2 weeks)! Let's do that

**Comment by xat** [comment]
we talked about it in the meantime in discord: probably easier to manually re-apply the changes that were made in the main branch in this branch. could become extremely painful otherwise :)

**Comment by Zir0h** [comment]
Can the [REACT_APP_IPFS_DEFAULT_GATEWAY](https://github.com/teia-community/teia-ui/blob/develop/.env#L20) also switch to CDN?

Maybe it's only on my end, but the calls to get the holder metadatafile are taking quite some time.
2022-11-04-145531_1579x542_scrot

Didn't we discuss using the indexer for that? Maybe that was implemented in one of the other PR's.

For example https://nftstorage.link/ipfs/QmXybiMjcZaJVLxyvCjPJWn1JbSmVsmtoBHfcxgN779p3C is also stored in the holder table:

```
query MyQuery {
  holder(where: {address: {_eq: "tz1XeEWsExj2BccfGifuTPcR6tEn6szjm4gQ"}}) {
    metadata
  }
}

{
  "data": {
    "holder": [
      {
        "metadata": {
          "identicon": "ipfs://QmfJ2EeaeX8ph2Q93VEcyeKcJDy1oNrJtvDNRc22pzyYzN",
          "description": "Artist"
        }
      }
    ]
  }
}
```

**Comment by xat** [comment]
@Zir0h this is fixed now in the latest commit.

**Pull request merged**